src/mds/Server.cc

```diff
@@ -1719,7 +1758,26 @@ void Server::submit_mdlog_entry(LogEvent *le, MDSLogContextBase *fin, MDRequestR
 void Server::respond_to_request(MDRequestRef& mdr, int r)
 {
   if (mdr->client_request) {
-    reply_client_request(mdr, MClientReply::create(*mdr->client_request, r));
+    MClientReply::ref reply = MClientReply::create(*mdr->client_request, r);
+    reply_client_request(mdr, reply);
```

Comment:
maybe we should call reply_client_request() after the batch code. because reply_client_request() drops locks

Reply:
yes, that's right. I'll change it:-)